### PR TITLE
[Feat][Connector-v2][Kafka] Add support for Schema Registry wire format in Protobuf deserialization

### PR DESCRIPTION
**Pull request by zhangshenghang**
… 

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If necessary, please update `incompatible-changes.md` to describe the incompatibility caused by this PR.
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)

### CODE REVIEW

**Review by chl-wxp — CHANGES_REQUESTED** [review]
```diff
@AutoService(Factory.class)
public class KafkaSourceFactory implements TableSourceFactory {

    @Override
    public String factoryIdentifier() {
        return "Kafka";
    }

    @Override
    public OptionRule optionRule() {
        return OptionRule.builder()
                .required(KafkaSourceOptions.BOOTSTRAP_SERVERS)
                .exclusive(
                        KafkaSourceOptions.TOPIC,
                        KafkaSourceOptions.TABLE_CONFIGS,
                        KafkaSourceOptions.TABLE_LIST)
                .optional(
                        KafkaSourceOptions.START_MODE,
+             .optional(KafkaSourceOptions.STRIP_SCHEMA_REGISTRY_HEADER)

**Comment by chl-wxp** [comment]
please add doc

**Comment by zhangshenghang** [comment]
@chl-wxp @davidzollo @corgy-w  ptal

**Review by davidzollo — DISMISSED** [review]
+1
LGTM

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceFactory.java [hunk 1]
```diff
@@ -59,7 +59,8 @@ public OptionRule optionRule() {
                         KafkaSourceOptions.DEBEZIUM_RECORD_TABLE_FILTER,
                         KafkaSourceOptions.KEY_PARTITION_DISCOVERY_INTERVAL_MILLIS,
                         KafkaSourceOptions.READER_CACHE_QUEUE_SIZE,
-                        KafkaSourceOptions.IGNORE_NO_LEADER_PARTITION)
+                        KafkaSourceOptions.IGNORE_NO_LEADER_PARTITION,
+                        KafkaSourceOptions.STRIP_SCHEMA_REGISTRY_HEADER)
```
Comment (chl-wxp) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceFactory.java]: The `STRIP_SCHEMA_REGISTRY_HEADER` parameter is only valid when FORMAT is `PROTOBUF`, so should `conditional()` be used?

File: seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    private final ProtobufDeserializationSchema inner;
+    private final SeaTunnelRowType rowType;
+
+    public SchemaRegistryAwareProtobufDeserializationSchema(CatalogTable catalogTable) {
+        this.inner = new ProtobufDeserializationSchema(catalogTable);
+        this.rowType = catalogTable.getSeaTunnelRowType();
+    }
+
+    @Override
+    public SeaTunnelRow deserialize(byte[] message) throws IOException {
```
Comment (chl-wxp) [inline seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java:64]: I have observed that there are a lot of `if else` in this method, and I suggest supplementing unit testing for the `deserialize()` method and adding more boundary condition tests.

File: seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+        int length = message.length;
+
+        // Confluent Schema Registry Protobuf wire format:
+        // 1 byte magic (0), 4 bytes schema id, N bytes message indexes (varints), then protobuf.
+        if (length >= 6 && message[0] == 0) {
+            int candidateStart = 6;
+            if (candidateStart < length) {
+                try {
+                    return inner.deserialize(Arrays.copyOfRange(message, candidateStart, length));
```
Comment (chl-wxp) [inline seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java]: Is `copyOfRange` risky when the amount of data is large and will it cause GC pressure? Of course, I am not sure if there is a better way to implement it. This is just a thought.

File: seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                } catch (IOException | RuntimeException ignored) {
+                    LOG.warn("Protobuf message not recognized at candidate offset 6, falling back");
```
Comment (chl-wxp) [inline seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java]: If `deserialize` fails, can an exception be thrown directly and the program terminated? Will just printing the warn log affect data consistency?

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceFactory.java (hunk 1 above)
Comment (zhangshenghang) [reply seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceFactory.java]: done

File: seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java (hunk 4 above)
Comment (zhangshenghang) [reply seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java]: Exceptions cannot be thrown here; this logic is correct and needs to continue executing.

File: seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java (hunk 3 above)
Comment (zhangshenghang) [reply seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java]: done

File: seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java (hunk 4 above)
Comment (chl-wxp) [reply seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/SchemaRegistryAwareProtobufDeserializationSchema.java]: > Exceptions cannot be thrown here; this logic is correct and needs to continue executing.

got it

**Review by davidzollo — DISMISSED** [review]
+1 if CI passed

**Review by chl-wxp — APPROVED** [review]
LGTM

**Review by davidzollo — APPROVED** [review]
+1
LGTM

**Review by hawk9821 — APPROVED** [review]
